pkg/apis/pipeline/v1/param_types_test.go

```diff
@@ -453,3 +454,41 @@ func TestArrayReference(t *testing.T) {
 		}
 	}
 }
+
+func TestExtractNames(t *testing.T) {
+	tests := []struct {
+		name   string
+		params v1.Params
+		want   sets.String
+	}{{
+		name:   "no params",
+		params: v1.Params{{}},
+		want:   sets.NewString(""),
+	}, {
+		name: "extract param names from ParamTypeString",
+		params: v1.Params{{
+			Name: "IMAGE", Value: v1.ParamValue{Type: v1.ParamTypeString, StringVal: "image-1"},
+		}, {
+			Name: "DOCKERFILE", Value: v1.ParamValue{Type: v1.ParamTypeString, StringVal: "path/to/Dockerfile1"}}},
+		want: sets.NewString("IMAGE", "DOCKERFILE"),
+	}, {
+		name: "extract param names from ParamTypeArray",
+		params: v1.Params{{
+			Name: "GOARCH", Value: v1.ParamValue{Type: v1.ParamTypeArray, ArrayVal: []string{"linux/amd64", "linux/ppc64le", "linux/s390x"}},
+		}},
+		want: sets.NewString("GOARCH"),
+	}, {
+		name: "extract param names from ParamTypeString and ParamTypeArray",
+		params: v1.Params{{
+			Name: "GOARCH", Value: v1.ParamValue{Type: v1.ParamTypeArray, ArrayVal: []string{"linux/amd64", "linux/ppc64le", "linux/s390x"}},
+		}, {
+			Name: "IMAGE", Value: v1.ParamValue{Type: v1.ParamTypeString, StringVal: "image-1"},
+		}},
+		want: sets.NewString("GOARCH", "IMAGE"),
+	}}
+	for _, tt := range tests {
```

Comment:
can you add a test for duplicate param names?
(TBH I'm sort of confused why you'd like to dedup, I would imagine duplicate params would be an error)

Reply:
@lbernick Added a test for duplicates. Checking for duplicate parameters within a Matrix is handled in a separate function `validateParams()`. This function `validateParameterInOneOfMatrixOrParams()` is checking that there aren't duplicates between PT Parameters and Matrix Parameters, so we can ignore duplicates in Matrix Parameters.

Reply:
Indeed I did 🤦‍♀️